packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -120,54 +133,139 @@ export type EditableTreeOrPrimitive = EditableTree | PrimitiveValue;
+     * It reads all nodes at once before the iteration starts to get a "snapshot" of this field.
+     * It might be inefficient regarding resources, but avoids situations
+     * when the field is getting changed while iterating.
+     */
+    [Symbol.iterator](): IterableIterator<UnwrappedEditableTree>;
+}
 
 /**
  * Unwrapped field.
  * Non-sequence multiplicities are unwrapped to the child tree or `undefined` if there is none.
- * Sequence multiplicities are handled with arrays.
+ * Sequence multiplicities are handled with {@link EditableField}.
  * See {@link UnwrappedEditableTree} for how the children themselves are unwrapped.
  */
-export type UnwrappedEditableField =
-    | UnwrappedEditableTree
-    | undefined
-    | readonly UnwrappedEditableTree[];
+export type UnwrappedEditableField = UnwrappedEditableTree | undefined | EditableField;
 
-export class ProxyTarget {
+/**
+ * This is a base class for `ProxyTarget` and `SequenceProxyTarget`, which uniformly handles cursors and anchors.
+ */
+export abstract class BaseProxyTarget {
     private readonly lazyCursor: ITreeSubscriptionCursor;
-    private anchor?: Anchor;
-
-    constructor(public readonly context: ProxyContext, cursor: ITreeSubscriptionCursor) {
-        this.lazyCursor = cursor.fork();
-        context.withCursors.add(this);
+    private anchor?: Anchor | FieldAnchor;
+
+    constructor(
+        public readonly context: ProxyContext,
+        cursor: ITreeSubscriptionCursor,
+        public readonly fieldKey: FieldKey,
+    ) {
+        if (this.fieldKey === rootFieldKeySymbol && cursor.mode === CursorLocationType.Fields) {
+            this.lazyCursor = this.context.forest.allocateCursor();
+            moveToDetachedField(this.context.forest, this.lazyCursor);
+        } else {
+            this.lazyCursor = cursor.fork();
+        }
+        this.context.withCursors.add(this);
     }
 
     public free(): void {
         this.lazyCursor.free();
         this.context.withCursors.delete(this);
         if (this.anchor !== undefined) {
-            this.context.forest.anchors.forget(this.anchor);
+            if (this.isFieldProxyTarget) {
+                const fieldAnchor = this.anchor as FieldAnchor;
+                if (fieldAnchor.parent !== undefined) {
+                    this.context.forest.anchors.forget(fieldAnchor.parent);
+                }
+            } else {
+                this.context.forest.anchors.forget(this.anchor as Anchor);
+            }
             this.context.withAnchors.delete(this);
             this.anchor = undefined;
         }
     }
 
-    public getAnchor(): Anchor {
+    public getNodeAnchor(): Anchor {
         if (this.anchor === undefined) {
             this.anchor = this.lazyCursor.buildAnchor();
             this.context.withAnchors.add(this);
         }
-        return this.anchor;
+        return this.anchor as Anchor;
+    }
+
+    public getFieldAnchor(): FieldAnchor {
```

Comment:
This is only used in one place: I think it would be better to just implement this in the subclass where it is used.

Same for getNodeAnchor.

Reply:
Reworked, please, see below.